### PR TITLE
Fix PVS-Studio static analyzer issues

### PR DESCRIPTION
**Pull request by zhaixuejun1993**
### Details:
 - Static analyzer issues: https://pvs-studio.com/en/blog/posts/cpp/1137/

 - *...*

### Tickets:
 - https://github.com/openvinotoolkit/openvino/issues/25247

### CODE REVIEW

File: src/plugins/intel_cpu/src/emitters/plugin/x64/jit_eltwise_emitters.cpp [hunk 1]
```diff
@@ -1655,7 +1655,7 @@ void jit_power_static_emitter::emit_isa(const std::vector<size_t> &in_vec_idxs,
         h->add(h->rsp, (get_max_vecs_count() + 2) * get_vec_length());
 
         // restore k registers
-        if (isa == x64::avx512_core || isa == x64::avx512_core) {
```
Comment (ilya-lavrenov) [inline src/plugins/intel_cpu/src/emitters/plugin/x64/jit_eltwise_emitters.cpp:1658]: @a-sidorova looks like we still need to handle different architecture for else branch? What should be the proper arch here?

File: src/plugins/intel_cpu/src/node.cpp [hunk 2]
```diff
@@ -396,7 +396,7 @@ void Node::resolveInPlaceEdges(Edge::LOOK look) {
 MemoryDescPtr Node::getBaseMemDescAtInputPort(size_t portNum) const {
     if (auto primDesc = getSelectedPrimitiveDescriptor()) {
         const auto& inConfs = primDesc->getConfig().inConfs;
-        if (inConfs.size() < portNum) {
+        if (portNum >= inConfs.size()) {
             OPENVINO_THROW("Can't get input memory desc at port: ", portNum, ", incorrect port number");
         }
```
Comment (ilya-lavrenov) [inline src/plugins/intel_cpu/src/node.cpp:401]: Can we use OPENVINO_ASSERT instead of if with throw?

File: src/plugins/intel_cpu/src/node.cpp [hunk 3]
```diff
@@ -407,7 +407,7 @@ MemoryDescPtr Node::getBaseMemDescAtInputPort(size_t portNum) const {
 MemoryDescPtr Node::getBaseMemDescAtOutputPort(size_t portNum) const {
     if (auto primDesc = getSelectedPrimitiveDescriptor()) {
         const auto& outConfs = primDesc->getConfig().outConfs;
-        if (outConfs.size() < portNum) {
+        if (portNum >= outConfs.size()) {
```
Comment (ilya-lavrenov) [inline src/plugins/intel_cpu/src/node.cpp:410]: the same

File: src/plugins/intel_cpu/src/nodes/bin_conv.cpp [hunk 4]
```diff
@@ -1085,7 +1085,7 @@ void BinaryConvolution::createPrimitive() {
     int r_pad_no_tail = nstl::max(0, (jcp.ow - jcp.ur_w_tail - 1) * jcp.stride_w
                                      + (jcp.kw - 1) * (jcp.dilate_w + 1) - (jcp.iw + jcp.l_pad - 1));
 
-    bool args_ok = jcp.l_pad <= jcp.ur_w && (r_pad_no_tail <= jcp.ur_w) && (jcp.l_pad <= jcp.ur_w) &&
```
Comment (ilya-lavrenov) [inline src/plugins/intel_cpu/src/nodes/bin_conv.cpp:1088]: @maxnick please, check

File: src/plugins/intel_cpu/src/emitters/plugin/x64/jit_eltwise_emitters.cpp (hunk 1 above)
Comment (a-sidorova) [reply src/plugins/intel_cpu/src/emitters/plugin/x64/jit_eltwise_emitters.cpp:1658]: This code part means the following: firstly we push all registers to stack to save the original values, then make function call and after all we pop all values to the corresponding registers from stack. It's classical task `preamble` and `postamle`. AVX512 also provides mask registers (which are missed on sse42 and avx2) so we have to save them in the stack too.
So we don't need to handle other isa here.
Thank you!

@zhaixuejun1993 FYI: By the way Snippets team implemented the helpers for internal call [preamble](https://github.com/BHbean/openvino/blob/748a751bf3854344fdb09e74d77188ce2de7e8e7/src/plugins/intel_cpu/src/emitters/plugin/x64/jit_emitter.cpp#L217) and [postamble](https://github.com/BHbean/openvino/blob/748a751bf3854344fdb09e74d77188ce2de7e8e7/src/plugins/intel_cpu/src/emitters/plugin/x64/jit_emitter.cpp#L251). In general we can reuse them here but I'm not sure that we should do it in this PR.

Comment (zhaixuejun1993) [reply src/plugins/intel_cpu/src/emitters/plugin/x64/jit_eltwise_emitters.cpp:1658]: @a-sidorova Actually I just saw the condition_1 is the same as the condition_2 in the "if( condition_1 || condition_2)", so I removed one.

Comment (a-sidorova) [reply src/plugins/intel_cpu/src/emitters/plugin/x64/jit_eltwise_emitters.cpp:1658]: Yes, I see. A few years ago this condition was `avx512_core && avx512_common`. After migration to new version of oneDNN we have `avx512_core && avx512_core`. So everything is ok

**Review by maxnick — APPROVED** [review]
The changes in the CPU plugin look good to me.

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**